plotnine/geoms/annotation_logticks.py

```diff
@@ -0,0 +1,139 @@
+            x_is_log_10, y_is_log_10 = y_is_log_10, x_is_log_10
+        result = []
+        if (not x_is_log_10 and ('t' in sides or 'b' in sides)):
+            warnings.warn("annotation_logticks for x-axis,"
+                          " but x-axis is not log10 - nonsensical marks show.",
+                          PlotnineWarning)
+        if (not y_is_log_10 and ('l' in sides or 'r' in sides)):
+            warnings.warn("annotation_logticks for y-axis,"
+                          " but y-axis is not log10 - nonsensical marks show.",
+                          PlotnineWarning)
+        return result
+
+    @classmethod
+    def draw_panel(cls, ignored_data, panel_params, coord, ax, **params):
+        if 'l' in params['sides'] or 'r' in params['sides']:
+            ymin, ymax = panel_params['y_range']
+            cls.draw_group('y', ymin, ymax, panel_params, coord, ax, params)
+        if 'b' in params['sides'] or 't' in params['sides']:
+            xmin, xmax = panel_params['x_range']
+            cls.draw_group('x', xmin, xmax, panel_params, coord, ax, params)
+
+    @staticmethod
+    def _calc_ticks(lower_log, upper_log, marks_at):
+        """Calculate log10 tick marks between two logs
+        (input is -1..4, not 0.1..10000).
+        Marks are set at the percentages defined by marks_at
+        ([0..1, ...]])
+        """
+        result = []
+        for l in range(lower_log, upper_log + 1):
+            for m in marks_at:
+                result.append(10**(l + np.log10(m)))
+        return result
+
+    @classmethod
+    def draw_group(cls, axis, min_value, max_value, panel_params, coord, ax,
+                   params):
+        cls._check_log_scale(params['sides'], panel_params['scales'],
+                             isinstance(coord, coord_flip))
+
+        lower_log = int(math.floor(min_value))
+        upper_log = int(math.ceil(max_value)) + 1
+        major = np.log10(cls._calc_ticks(lower_log, upper_log, [1.0]))
+        middle = np.log10(cls._calc_ticks(lower_log, upper_log, [0.5]))
+        minor = np.log10(cls._calc_ticks(lower_log, upper_log,
+                         [0.2, 0.3, 0.4, 0.6, 0.7, 0.8, 0.9]))
+
+        lengths = params.get('length', cls.DEFAULT_PARAMS['length'])
+        if len(lengths) != 3:
+            raise ValueError(
+                "length for annotation_logticks must be a tuple of 3 floats")
+
+        if 'length' in params:
+            del params['length']
+
+        for (positions, length) in [
+            (major, lengths[0]),
+            (middle, lengths[1]),
+            (minor, lengths[2]),
+        ]:
+            data = pd.DataFrame({
+                axis: positions,
+                'size': params['size'],
+                'color': params['color'],
+                'alpha': params['alpha'],
+                'linetype': params['linetype'],
+            })
+            super().draw_group(
+                data, panel_params, coord, ax, length=length, **params)
+
```

Comment:
The functionality in this method can be moved to `draw_panel`.

Reply:
It could, I'd rather not .
It would require a temporary list of tuples and a loop and an unintuitive control flow compared to two ifs+function calls (remember, you can have them on both axis in the same plot).